### ISSUE TITLE
correct interpretation / meaning of "file source ID"

### ISSUE THREAD

**Issue by rapidlasso** [ASPRSorg/LAS#59]
The specification does NOT say the the "file source ID" and the "point source ID" MUST be related. Should it say that? My understanding so far was that - at least for airborne / mobile / terrestrial LiDAR - the "file source ID" should be non-zero for individual flightlines / trajectories / scan positions and  zero for tiles (who instead use the "point source ID" to specify which flightline / trajectory / scan position a point originates from). The issue was raised [here](https://groups.google.com/d/topic/lastools/VdrUVCsoNkg/discussion).

**Comment by hobu** [comment]
I agree that the intention of "file source ID" and "point source ID" is the linkage of multi-file collections.

I disagree they MUST be connected. While that linkage is one prominent data model that is often constructed, it's not the only one. 

What's the case for making it be required?

**Comment by rapidlasso** [comment]
A big problem in LiDAR deliveries - as far as I have seen - is the improper use of these fields. Usually the point source IDs are the ones that are not correctly populated and thereby prohibiting a proper flightline alignment / overlap check to be done. When I wrote the [LASvalidator](http://github.com/LASvalidator) for the ASPRS I tried to make its checks follow the intended meaning of the fields of the LAS format as closely as possible. Unfortunately - and here I can lament again - the LWG has had most of its history deleted ... including the discussions that led to the design of these fields. Lewis? Karl?

**Comment by jdnimetz** [comment]
It is (or was) a requirement within the 3DEP lidar base specification. However, with current 3DEP requirement for tiled data only, this is less of an issue. As Martin points out, file source ID should be 0 for tiled LAS data.
Two use cases:
(1) In pre 64-bit days, LBS required chopping swaths into </= 2 GB chunks. In those cases, it was necessary for file source and point source IDs to be identical to tie the swath together.
(2) Accessing flight line ID from file source ID is more efficient than accessing point source ID within the point records. This is no longer a good use case for the USGS for the reason I mention above.

**Comment by rapidlasso** [comment]
The question is: Do we want to allow the "file source ID" to be some random number in case 
(a) the "point source IDs" are all identical because it's a flightline?
(b) the "point source IDs" are different because it's a tile?

**Comment by hobu** [comment]
> Usually the point source IDs are the ones that are not correctly populated and thereby prohibiting a proper flightline alignment / overlap check to be done

These fields exist for linear mode acquisition scenarios. Not all data written in LAS fits that scenario. Tightening up this language adds confusion to the scenarios where they don't apply.

**Comment by jdnimetz** [comment]
Is there a good use case to keep the file source ID?

**Comment by rapidlasso** [comment]
I've seen many collects that properly use the "file source ID" to number their flightlines according to the flight plan or the driving route of their survey area. I can go either way with this issue but I'd like the folks that designed the fields to have a chance to weigh in. Lewis? Karl? Paul? ...?

**Comment by hobu** [comment]
> Is there a good use case to keep the file source ID?

It's there, and it's used successfully by the scenarios that need it. IMO, there's no reason to toss it, but to make its proper population a hard requirement when there are many LAS data storage scenarios where it doesn't really apply seems overly constraining.

**Comment by jdnimetz** [comment]
@hobu - _I disagree they MUST be connected. While that linkage is one prominent data model that is often constructed, it's not the only one._

I agree with this statement.

**Comment by rapidlasso** [comment]
@jstoker74 @esilvia @anayegandhi @lgraham-geocue any strong feelings about how "file source ID" and "point source ID" should be used for linear LiDAR, other LiDAR, and non-LiDAR points?

**Comment by esilvia** [comment]
I agree with @hobu that forcing the connection to PtSourceID is overly constraining. I've seen it used in other contexts such as marking individual scan positions for terrestrial scans, or differentiating between data sources (e.g., USGS 3DEP vs. OpenTopo), or (as in the example from from @rapidlasso) to differentiate between scanners as with SystemID.

However, I do think that the FileSourceID and PtSourceID descriptions would benefit from some clarification outside the context of airborne lidar so I'll take a stab at that for R14.

@rapidlasso This is certainly a useful check for many applications, so I wouldn't remove it entirely. Maybe a viable approach is to make that check optional via a CLI flag? Or the opposite: provide a flag to disable the check. That's what I've done in my tools, e.g., providing a flag to disable the warning about missing CRS or extra bytes in the LAS header. A succinct list of possible tests to enable/disable in the lasvalidate_readme would work nicely.

**Comment by esilvia** [comment]
Does this capture what we're looking for?

### File Source ID
This field should be set to a value ranging from 1 to 65,535. A value of zero is interpreted to mean that an ID has not been assigned, which is often the norm for a LAS file resulting from a merge operation such as a tile.

Note that this scheme allows a LIDAR project to contain up to 65,535 unique sources. Example sources can include a data repository ID or an original collection such as a flight line for direct measurement airborne systems, a route number for mobile systems, or a setup identifier for static systems.

### Point Source ID
This value indicates the source from which this point originated, such as a swath number for conventional airborne systems, a route number for mobile systems, or a setup identifier for static systems. For a LAS file created during a merge operation, the Point Source ID could also equal the File Source ID of the origin LAS file.

Valid values for this field are 1 to 65,535 inclusive. Zero is reserved as a convenience to system implementers. A Point Source ID of zero implies that this point originated in this file, such as for a Synthetic point, or is derived from multiple sources.

**Comment by esilvia** [comment]
(original for comparison)
### File Source ID (original)
This field should be set to a value ranging from 1 to 65,535. If this file was derived from an original
flight line, this is often the flight line number. A value of zero (0) is interpreted to mean that an
ID has not been assigned. In this case, processing software is free to assign a number. Note that this scheme allows a LIDAR project to contain up to 65,535 unique sources. A source can be an
original flight line or it can be result of merge and/or extract operations.

### Point Source ID (original)
This value indicates the file from which this point originated. Valid values for this field are 1 to
65,535 inclusive with zero being used for a special case discussed below. The numerical value corresponds
to the File Source ID from which this point originated. Zero is reserved as a convenience
to system implementers. A Point Source ID of zero implies that this point originated in this file.
This implies that processing software should set the Point Source ID equal to the File Source ID
of the file containing this point at some time during processing.

**Comment by rapidlasso** [comment]
I see nothing wrong with the new formulation. Definitely more understandable than the old one.

**Comment by esilvia** [comment]
Thanks Martin! I've incorporated that change with a very slight refinement from what I posted earlier. 

Specifically, I started the FileSourceID by stating it can be a value from **0** to 65,535, not **1** to 65,535. I also omitted "often" from the first paragraph since it's redundant. Same for "also" in the first paragraph of the PtSourceID.

PDF here: https://s3.amazonaws.com/asprs-las/LAS-specification-25834cf92b54e5d7e4c00df055469b064ceb83da.pdf

![image](https://user-images.githubusercontent.com/11743012/48083906-3f275680-e1ab-11e8-9a0d-9a9a7c43862b.png)

![image](https://user-images.githubusercontent.com/11743012/48084026-86154c00-e1ab-11e8-85bb-d6127e371c08.png)
